marker opacity for scatter plot viz

moontrip opened the pull request:
This will address https://github.com/VEuPathDB/web-eda/issues/1487. Relying on https://github.com/VEuPathDB/web-components/pull/419. This also works with overlay and facet variables: tested but no screenshot is attached. 

Note that currently, the step of opacity is set to be 0.1.

Here are some screenshots:
- a general marker with opacity of 0
 ![scatter-viz-opacity0](https://user-images.githubusercontent.com/12802305/204834443-ec1626d3-a5d5-446e-8c68-84bc205c7645.png)

- a general marker with opacity of 1
![scatter-viz-opacity1](https://user-images.githubusercontent.com/12802305/204834475-b8ff0fcf-e855-49e5-bfcf-b643e2a1fbb2.png)

- a gradient marker with opacity of 0
![scatter-viz-gradient-opacity0](https://user-images.githubusercontent.com/12802305/204834509-bb429337-a037-42f7-b450-b904de55ccb5.png)

- a gradient marker with opacity of 1
![scatter-viz-gradient-opacity1](https://user-images.githubusercontent.com/12802305/204834530-ca02af5d-0a6a-431d-ac1d-f413a6c35b6e.png)

bobular commented on src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:
```diff
@@ -2246,6 +2276,15 @@ function processInputData<T extends number | string>(
             seriesGradientColorscale?.length > 0
               ? markerSymbolGradient
               : markerSymbol(index),
+          // need to set marker.line for a transparent case (opacity != 1)
```
The `opacity: 0.7` a few lines above here is being applied to the final plot, so when the user chooses 1 on the slider, they actually only get 0.7.

The solution may just to remove this line.

moontrip replied:
@bobular Thank you for your suggestion. I will remove it! 👍

bobular reviewed (changes requested):
It looks/works great! The only things missing are
a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec)
b) opacity should be added to thumbnail dependencies

moontrip commented:
> It looks/works great! The only things missing are a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec) b) opacity should be added to thumbnail dependencies

@bobular Thank you for your valuable suggestions! 👍 I missed those parts at Viz. Will work for that, probably tomorrow and after done, I will certainly use "re-review request" icon 😉

moontrip commented:
> It looks/works great! The only things missing are a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec) b) opacity should be added to thumbnail dependencies

@bobular I have addressed your suggestions. Since the corresponding web-components PR was already merged, you can test this PR without using npm-pack-here. But please don't forget to update your local repo (git pull origin scatterplot-opacity & yarn install) 😉

bobular commented:
Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name?  Perhaps `markerBodyOpacity` or `markerCenterOpacity`?

moontrip commented on src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:
```diff
@@ -1220,7 +1232,7 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
       : undefined,
     // need to define markerColorOpacity for faceted plot
-    markerColorOpacity: markerColorOpacity,
+    markerColorOpacity: vizConfig.markerOpacity,
```
@bobular Ah, yes. You are absolutely right. I will start working on this PR after the other work you commented/suggested, including new spec such as two banners etc. 👍

moontrip commented:
> Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name? Perhaps `markerBodyOpacity` or `markerCenterOpacity`?

@bobular thank you for your suggestion. Perhaps markerBodyOpacity may work? I will need to change it at web-components as well 😄

bobular commented:
> > Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name? Perhaps `markerBodyOpacity` or `markerCenterOpacity`?
> 
> @bobular thank you for your suggestion. Perhaps markerBodyOpacity may work? I will need to change it at web-components as well 😄

Oh, that's a pain re. web-components.  Maybe find a consistent name on the EDA-side that means no changes over there?

moontrip commented:
> markerBodyOpacity

@bobular well, the current prop name, markerColorOpacity, is defined in web-components so it should be changed in there anyway. It may not be a big deal, except you will have one additional review request 😄

moontrip commented:
@bobular I have addressed your feedbacks. Since the change of the prop name required to make a change at web-components, this PR now depends on https://github.com/VEuPathDB/web-components/pull/433.

moontrip commented:
@bobular update. Due to fixing a bug for marker opacity work at web-components, https://github.com/VEuPathDB/web-components/pull/433 is now merged owing to Jeremy. So from now one this PR does not have a dependency with web-components.

bobular reviewed (approved):
This looks good - I'm going to approve without testing because I'm having trouble getting my dev instance up, and will focus my testing on #1464 when that is sorted

moontrip commented:
@bobular Thank you for your reviews 👍
